docs/concepts/interface-driven-development.mdx

```diff
@@ -3,39 +3,34 @@ title: "Interface Driven Development"
 date: 2020-05-01T00:00:00+00:00
 icon: "ti-anchor" # themify icon pack : https://themify.me/themify-icons
 description: "A quick guide to interface-driven development"
-sidebar_position: 1
+sidebar_position: 5
 type: "docs"
 ---
 
-Interface Driven Development (**IDD**), aka contract-driven development,
-is a development approach that _focuses on defining **what** components
-need before **how** to meet those needs_.
+## Overview
 
-**Systems developed using IDD - especially
-distributed systems - are _more loosely coupled, robust and maintainable_.**
+Interface Driven Development (**IDD**), aka contract-driven development, is a development approach that _focuses on defining **what** components need before **how** to meet those needs_.
 
-WebAssembly modules operate in a completely isolated sandbox, meaning
-that they can only perform logical operations with no access to system
-resources like I/O, networking and syscalls.
+Systems developed using IDD — especially distributed systems — are _more loosely coupled, robust and maintainable_.
 
-In functional programming terminology we might call these modules "pure",
-as they can only map inputs to outputs without producing side-effects.
-Without any side-effects, how can we use WebAssembly components to do anything
-useful in our applications?
+By default, [WebAssembly components](./webassembly-components.mdx) operate in a completely isolated sandbox, meaning that they can only perform logical operations with no access to system resources like I/O, networking, and syscalls.
 
-Interface (or contract) driven development is the core that makes wasmCloud possible.
+In functional programming terminology we might call these modules "pure", as they can only map inputs to outputs without producing side-effects. 
 
-In order to use isolated WebAssembly modules for useful work, we have to build
-explicit paths in and out of the module and pass them as inputs.
-These paths are defined using abstractions called _interfaces_ which specify
-exactly what kinds of functions and data types the modules have access to.
+But without any side-effects, how can we use WebAssembly components to do anything useful in our applications?
 
-These interfaces give our sandboxed WebAssembly components access to *external*
-services:
-- key-value stores
-- HTTP servers & clients
-- message brokers
-- ... and much more
+Interface driven development is the core that makes wasmCloud possible. 
```

Comment:
Is this opportunity to talk about WASI?

Reply:
That's a good idea but 🧠 is mush so will punt on this